src/main/ruby/core/truffle/cext.rb

```diff
@@ -141,6 +141,9 @@ def size
     end
 
     def to_native
+      # Hold on to a duplicate of the string to prevent its rope being
+      # GC'ed while we might still need it.
+      Truffle::CExt.rb_tr_new_local_ref(@string.dup)
       Rubinius::FFI::Pointer.new(Truffle::CExt.string_pointer_to_native(@string))
```

Comment:
Did you have such a case?
In MRI, the RSTRING_PTR is only valid while holding the Ruby String instance (for instance with RB_GC_GUARD), I think we should be able to play the same trick, and that would be less expensive.